profiles/source_profile.go

```diff
@@ -805,8 +806,25 @@ func NewSourceProfile(s string, source string, n NewSourceProfileInterface) (Sou
 	} else if format, ok := params["format"]; ok {
 		// File is not passed in from stdin or specified using "file" flag.
 		return SourceProfile{Ty: SourceProfileTypeFile}, fmt.Errorf("file not specified, but format set to %v", format)
-	} else if file, ok := params["config"]; ok {
+	} else if file, ok := params["shardingConfig"]; ok {
 		config, err := n.NewSourceProfileConfig(strings.ToLower(source), file)
+		shardIdConfig, ok2 := params["shardIdConfig"]
+		if ok2 {
+			switch shardIdConfig {
+			case constants.SHARD_ID_PREFIX:
+				config.ShardIdConfig = shardIdConfig
+			case constants.SHARD_ID_SUFFIX:
+				config.ShardIdConfig = shardIdConfig
+			case constants.SHARD_ID_NONE:
+				config.ShardIdConfig = shardIdConfig
+			default:
+				return SourceProfile{}, fmt.Errorf("invalid shard id config passed")
+			}
+		} else {
+			//Default to setting shard id config to none
+			config.ShardIdConfig = constants.SHARD_ID_NONE
+		}
```

Comment:
`n.NewSourceProfileConfig` seems to be a misnomer now since it is supposed to return a `SourceProfileConfig` but the `SourceProfileConfig` is now actually a combination of `shardIdConfig` and `ShardConfiguration*` (bulk, dms or dataflow). The `shardIdConfig` only gets added in later..

We should ideally rename `NewSourceProfileConfig` to something like `parseShardConfigurationFile` or shift this logic inside it so that it still returns the completed `SourceProfileConfig` implementation.